src/components/modal/v2/lib/hooks/currency.js

```diff
@@ -10,6 +10,7 @@ export const currencyFormat = string => {
             // eslint-disable-next-line security/detect-unsafe-regex
             /((\$|£)?(\d{1,5}(\.|,)){1,3}00)/g
```

Comment:
```suggestion
            /((\$|£)?(\d{1,5}(\.|,)){1,3}00)(\s?EUR)?/g
```
If we update this regex here it should cover our usecase.